wman-backend/Wman.Logic/Classes/EventLogic.cs

```diff
@@ -69,7 +74,10 @@ public async Task MassAssignUser(int eventID, ICollection<string> usernames)
                 {
                     throw new ArgumentException($"User: {0} not found", item);
                 }
-
+                if (await userManager.IsInRoleAsync(selectedUser, "Worker") == false)
```

Comment:
Same advice applies here.
```suggestion
                if (!await userManager.IsInRoleAsync(selectedUser, "Worker"))
```